### DISCUSSION TITLE
debug info pollutes terminal when using mpv or other terminal apps

### DISCUSSION THREAD

**Discussion by deewakar** [Arkq/bluez-alsa#568] [Q&A] [answered]
I'm using bluealsa inside debian/testing and installed it with the package manager. When playing an audio file with mpv, stdout is cluttered with several debug messages like this:

D: bluealsa-pcm.c:895: Getting BlueALSA PCM: PLAYBACK 9A:BC:04:06:66:FC a2dp
D: bluealsa-pcm.c:769: /org/bluealsa/hci0/dev_9A_BC_04_06_66_FC/a2dpsrc/sink: Setting constraints
D: bluealsa-pcm.c:396: /org/bluealsa/hci0/dev_9A_BC_04_06_66_FC/a2dpsrc/sink: Initializing HW
D: bluealsa-pcm.c:416: /org/bluealsa/hci0/dev_9A_BC_04_06_66_FC/a2dpsrc/sink: FIFO buffer size: 4096
D: bluealsa-pcm.c:422: /org/bluealsa/hci0/dev_9A_BC_04_06_66_FC/a2dpsrc/sink: Selected HW buffer: 4 periods x 4800 bytes == 19200 bytes
D: bluealsa-pcm.c:440: /org/bluealsa/hci0/dev_9A_BC_04_06_66_FC/a2dpsrc/sink: Initializing SW
D: bluealsa-pcm.c:440: /org/bluealsa/hci0/dev_9A_BC_04_06_66_FC/a2dpsrc/sink: Initializing SW
D: bluealsa-pcm.c:440: /org/bluealsa/hci0/dev_9A_BC_04_06_66_FC/a2dpsrc/sink: Initializing SW
D: bluealsa-pcm.c:477: /org/bluealsa/hci0/dev_9A_BC_04_06_66_FC/a2dpsrc/sink: Prepared
D: bluealsa-pcm.c:440: /org/bluealsa/hci0/dev_9A_BC_04_06_66_FC/a2dpsrc/sink: Initializing SW
D: bluealsa-pcm.c:477: /org/bluealsa/hci0/dev_9A_BC_04_06_66_FC/a2dpsrc/sink: Prepared
D: bluealsa-pcm.c:316: /org/bluealsa/hci0/dev_9A_BC_04_06_66_FC/a2dpsrc/sink: Starting
D: bluealsa-pcm.c:176: /org/bluealsa/hci0/dev_9A_BC_04_06_66_FC/a2dpsrc/sink: Waiting for first period of frames
D: bluealsa-pcm.c:183: /org/bluealsa/hci0/dev_9A_BC_04_06_66_FC/a2dpsrc/sink: Starting IO loop: 11
D: bluealsa-pcm.c:349: /org/bluealsa/hci0/dev_9A_BC_04_06_66_FC/a2dpsrc/sink: Stopping
D: bluealsa-pcm.c:129: /org/bluealsa/hci0/dev_9A_BC_04_06_66_FC/a2dpsrc/sink: IO thread cleanup
D: bluealsa-pcm.c:477: /org/bluealsa/hci0/dev_9A_BC_04_06_66_FC/a2dpsrc/sink: Prepared

How do I stop these messages from cluttering my terminal? Thanks.

**Comment by arkq** [comment, marked as answer]
The debug messages are printed into the standard error. So, you can try to redirect stderr to `/dev/null`, i.e.: `mpv file.mp4 2>/dev/null`. However, you will loose stderr output from mpv as well.

I'm not sure what is the policy in Debian testing about debug builds... Configuring bluez-alsa with `--enable-debug` is not a release build, it's a debug build with extra debugging and assertions. Maybe you should ask bluez-alsa deb package maintainer to rebuild this package without `--enable-debug` option.

**Comment by deewakar** [comment]
Redirecting stderr works but I'd have to do that for every terminal audio app I use. Is there a way to stop this at the source, maybe something I can include in my systemd service file?

> Configuring bluez-alsa with --enable-debug is not a release build, it's a debug build with extra debugging and assertions. Maybe you should ask bluez-alsa deb package maintainer to rebuild this package without --enable-debug option.

Sorry if I'm misinterpreting, but are these messages the result of enabling the `--enable-debug` option? I might contact bluez-alsa package maintainer if that's the case. Thank you.

**Reply by arkq** [reply]
>Redirecting stderr works but I'd have to do that for every terminal audio app I use.

You can use simple wrapper (place it on the PATH, but before `/usr/bin` or wherever the `mpv` is) for that, e.g.:

```sh
#!/bin/sh
exec /usr/bin/mpv "$@" 2>/dev/null
```

> but are these messages the result of enabling the `--enable-debug` option? 

Yes, these messages are the result of configuration option `--enable-debug`. You can not disable/enable it at runtime - there is no such option in bluez-alsa for that.
